### PR TITLE
Template part block - fix PHP notice for placeholder block.

### PR DESCRIPTION
**Pull request by Addison-Stavlo**
## Description

When we attempt to load a post containing a template part block placeholder block a php notice is present:

Screen Shot 2021-04-16 at 6 49 13 PM

Or on the front-end:

Screen Shot 2021-04-16 at 6 54 02 PM

This is because when no content for the template part is resolved, a message is printed to notify the user using `sprintf` to attach the `slug` attribute of the template part to the message:

Screen Shot 2021-04-16 at 6 53 43 PM

Since a template part placeholder block has no `slug` attribute, this also causes `sprintf` to throw the php notice.  However, we should not be showing this message if no `slug` attribute is present since absence of this attribute indicates a placeholder which by definition there should be no corresponding content found.

In this PR we return nothing in the case of the placeholder, since no content is expected and no error message is required.

## How has this been tested?

* Add a template part placeholder block to the post editor (this can be done by selecting 'template part' in the block inserter).
* Save the post.
* preview the post on the front-end.
* verify there is no longer a visible php notice or corresponding message where the template part should be.

## Checklist:
- [ ] My code is tested.
- [ ] My code follows the WordPress code style. 
- [ ] My code follows the accessibility standards. 
- [ ] I've tested my changes with keyboard and screen readers. 
- [ ] My code has proper inline documentation. 
- [ ] I've included developer documentation if appropriate. 
- [ ] I've updated all React Native files affected by any refactorings/renamings in this PR (please manually search all `*.native.js` files for terms that need renaming or removal).

### CODE REVIEW

File: packages/block-library/src/template-part/index.php [hunk 1]
```diff
@@ -19,12 +19,7 @@ function render_block_core_template_part( $attributes ) {
 	$content          = null;
 	$area             = WP_TEMPLATE_PART_AREA_UNCATEGORIZED;
 
-	if ( ! empty( $attributes['postId'] ) && get_post_status( $attributes['postId'] ) ) {
-		$template_part_id = $attributes['postId'];
-		// If we have a post ID and the post exists, which means this template part
-		// is user-customized, render the corresponding post content.
-		$content = get_post( $attributes['postId'] )->post_content;
-	} elseif (
```
Comment (Addison-Stavlo) [inline packages/block-library/src/template-part/index.php:27]: Note - removing this section is unrelated to the error noted in the PR description.

This code block is no longer necessary since `postId` has not been a valid attribute of the template part block for months now and content is always found via the `slug`/`theme` attributes.

**Review by carolinan — APPROVED** [review]
Assuming that with a template part placeholder, you mean a template part where one has not chosen whether to add a new or existing template part, then I am able to reproduce the PHP notice. Applying the PR solves it.

**Review by david-szabo97 — APPROVED** [review]
Works as expected! 🚢
